go/tasks/plugins/k8s/dask/dask.go

```diff
@@ -0,0 +1,433 @@
+)
+
+type defaults struct {
+	Namespace   string
+	Image       string
+	Container   v1.Container
+	Resources   *v1.ResourceRequirements
+	Env         []v1.EnvVar
+	Annotations map[string]string
+}
+
+func getDefaults(ctx context.Context, taskCtx pluginsCore.TaskExecutionContext, taskTemplate core.TaskTemplate) (*defaults, error) {
+	executionMetadata := taskCtx.TaskExecutionMetadata()
+
+	defaultContainerSpec := taskTemplate.GetContainer()
+	if defaultContainerSpec == nil {
+		return nil, errors.Errorf(errors.BadTaskSpecification, "task is missing a default container")
+	}
+
+	defaultImage := defaultContainerSpec.GetImage()
+	if defaultImage == "" {
+		return nil, errors.Errorf(errors.BadTaskSpecification, "task is missing a default image")
+	}
+
+	envVars := []v1.EnvVar{}
+	if taskTemplate.GetContainer().GetEnv() != nil {
+		for _, keyValuePair := range taskTemplate.GetContainer().GetEnv() {
+			envVars = append(envVars, v1.EnvVar{Name: keyValuePair.Key, Value: keyValuePair.Value})
+		}
+	}
+
+	var defaultResources *v1.ResourceRequirements
+	if executionMetadata.GetOverrides() != nil && executionMetadata.GetOverrides().GetResources() != nil {
+		defaultResources = executionMetadata.GetOverrides().GetResources()
+	}
+
+	// The default container will be used to run the job (i.e. within the driver pod)
+	defaultContainer := v1.Container{
+		Name:  "job-runner",
+		Image: defaultImage,
+		Args:  defaultContainerSpec.GetArgs(),
+		Env:   envVars,
+	}
+
+	templateParameters := template.Parameters{
+		TaskExecMetadata: taskCtx.TaskExecutionMetadata(),
+		Inputs:           taskCtx.InputReader(),
+		OutputPath:       taskCtx.OutputWriter(),
+		Task:             taskCtx.TaskReader(),
+	}
+	err := flytek8s.AddFlyteCustomizationsToContainer(ctx, templateParameters, flytek8s.ResourceCustomizationModeAssignResources, &defaultContainer)
+	if err != nil {
+		return nil, err
+	}
+
+	return &defaults{
+		Namespace:   executionMetadata.GetNamespace(),
+		Image:       defaultImage,
+		Container:   defaultContainer,
+		Resources:   defaultResources,
+		Env:         envVars,
+		Annotations: executionMetadata.GetAnnotations(),
+	}, nil
+}
+
+func createResourceList(pb_resources []*core.Resources_ResourceEntry) (v1.ResourceList, error) {
+	resourceList := v1.ResourceList{}
+	for _, resourceEntry := range pb_resources {
+		resourceName, ok := resourceMapping[resourceEntry.GetName()]
+		if !ok {
+			return nil, errors.Errorf(errors.BadTaskSpecification, fmt.Sprintf("Could not tranlate resource with name %s to k8s resource", resourceEntry.GetName()))
+		}
+		quantity, err := resource.ParseQuantity(resourceEntry.GetValue())
+		if err != nil {
+			return nil, errors.Errorf(errors.BadTaskSpecification, fmt.Sprintf("Could not parse %s as k8s resource", resourceEntry.GetValue()))
+		}
+		resourceList[resourceName] = quantity
+	}
+	return resourceList, nil
+}
+
+func convertProtobufResourcesToK8sResources(pb_resources *core.Resources) (*v1.ResourceRequirements, error) {
```

Comment:
probably a good idea to stick with camel-case per go standards - `pb_resources` -> `pbResources`

Reply:
@hamersaw Thanks for seeing this, the Python developer coming through! I ran `golangcci-lint` now - so should be all fixed 👍